### PR TITLE
fix: provide the metadata by the block heights

### PR DESCRIPTION
**Pull request by xsteadybcgo**
## Context

* [Need to provide metadata for a specific block height.](https://parallelfi.atlassian.net/browse/BTE-175). 
* This pr only query data from subquery directly to meet the demand. I'll pull another request to set **_cache_** to prevent service unavailability after the BTE-130 is merged

## Were any new external packages added?

* List external packages added and why they were added
No

### CODE REVIEW

**Review by crystalynliu — APPROVED** [review]

File: packages/app/src/controller/staking/metadata.ts [hunk 1]
```diff
@@ -63,3 +64,27 @@ export async function getParaStakingMetadataSnapshots(ctx: Context, next: Next)
...
+  const metadatas = await queryStakingMetadata(heikoGraphQLClient, queryBlocks)
+  ctx.body = Resp.Ok({
+    metadatas
+  })
+  return next()
+}
+
+export async function getParaStakingMetadataScope(ctx: Context, next: Next) {
+  const { heights } = ctx.request.query
+  const queryBlocks = heights ? (heights as string).split(',').map(i => Number(i)) : []
```
Comment (zhihuizhangjaden) [inline packages/app/src/controller/staking/metadata.ts:86]: Why no `heights` validation logic for para?

File: packages/app/src/controller/staking/metadata.ts [hunk 2]
```diff
@@ -63,3 +64,27 @@ export async function getParaStakingMetadataSnapshots(ctx: Context, next: Next)
   })
   return next()
 }
+
+export async function getHeikoStakingMetadataScope(ctx: Context, next: Next) {
+  const { heights } = ctx.request.query
+  if (!heights) {
+    throw Resp.Fail(Code.Param_Invalid, `heights is required parameters` as Msg)
+  }
+  if (typeof heights !== 'string') {
```
Comment (BruceChar) [inline packages/app/src/controller/staking/metadata.ts:73]: this check is redundant and what should to do is the format check, a number string split by `,`

File: packages/app/src/controller/staking/metadata.ts [hunk 3]
```diff
@@ -63,3 +64,27 @@ export async function getParaStakingMetadataSnapshots(ctx: Context, next: Next)
...
+  ctx.body = Resp.Ok({
+    metadatas
+  })
+  return next()
+}
+
+export async function getParaStakingMetadataScope(ctx: Context, next: Next) {
+  const { heights } = ctx.request.query
+  const queryBlocks = heights ? (heights as string).split(',').map(i => Number(i)) : []
+  const metadatas = await queryStakingMetadata(paraGraphQLClient, queryBlocks)
```
Comment (BruceChar) [inline packages/app/src/controller/staking/metadata.ts:87]: It's better that we handle the query result if error occurs. we should not respond the result directly.

File: packages/app/src/controller/staking/metadata.ts (hunk 2 above)
Comment (xsteadybcgo) [reply packages/app/src/controller/staking/metadata.ts:73]: It does need a format check. At a same time, we cannot guarantee that the user will pass only one heights, `ctx.request.query.heights` will be parsed as an `array`

File: packages/app/src/controller/staking/metadata.ts (hunk 3 above)
Comment (BruceChar) [reply packages/app/src/controller/staking/metadata.ts:87]: maybe after cache

File: packages/app/src/controller/staking/metadata.ts (hunk 2 above)
Comment (BruceChar) [reply packages/app/src/controller/staking/metadata.ts:73]: the query in which case is an array

File: packages/app/src/controller/staking/metadata.ts (hunk 3 above)
Comment (xsteadybcgo) [reply packages/app/src/controller/staking/metadata.ts:87]: Need cache, but BTE-130 has not been merged

File: packages/app/src/controller/staking/metadata.ts (hunk 1 above)
Comment (xsteadybcgo) [reply packages/app/src/controller/staking/metadata.ts]: Missing! Good catch

**Pull request merged**